alternator/executor.cc

```diff
@@ -137,8 +151,7 @@ static void validate_table_name(const std::string& name) {
         throw api_error::validation(
                 format("TableName must be at least 3 characters long and at most {} characters long", max_table_name_length));
     }
-    static const std::regex valid_table_name_chars ("[a-zA-Z0-9_.-]*");
-    if (!std::regex_match(name.c_str(), valid_table_name_chars)) {
+    if (!valid_table_name_chars(name)) {
```

Comment:
Do we need to validate the table name? If we only allow valid tables to be created, then we look up the table name, we are guaranteed it is valid.

Reply:
We'll still need to validate on table creation, but presumably those are less frequent than data access.

Reply:
> Do we need to validate the table name

Likely we don't need. This observation was mentioned by me and Nadav in https://github.com/scylladb/scylladb/pull/12445#issuecomment-1371467235 and https://github.com/scylladb/scylladb/pull/12445#issuecomment-1371485195

I propose to tackle this in a separate patch, the gain is probably much smaller and regression scope bigger. I created an issue: https://github.com/scylladb/scylladb/issues/12538

> And this is where we could also nicely print 'invalid name' errors to the users.

I'd say it's more about being "error compatible" than nice. If possible we should return the same error text as dynamo does. One of the reasons is that some application may (probably should not, but still...) act accordingly to this text. Other would be that it'd be easier to find those errors on stackoverflow :)